### ISSUE TITLE
new-app error handling of unreachable git repos is confusing

### ISSUE THREAD

**Issue by bparees** [openshift/origin#12917]
New-app tries to treat its primary argument as a source repo, but if it can't reach the repo, it falls back to treating it as a template, which leads to confusing errors like:

```
$ oc new-app https://githb.com/openshift/nodejs-e --dry-run
error: unable to load template file "https://githb.com/openshift/nodejs-e": Get https://githb.com/openshift/nodejs-e: dial tcp 158.69.145.50:443: i/o timeout
error: no match for "https://githb.com/openshift/nodejs-e"
```

or

```
$ oc new-app https://www.google.com/openshift/nodejs-e --dry-run
error: unable to load template file "https://www.google.com/openshift/nodejs-e": unable to read URL "https://www.google.com/openshift/nodejs-e", server reported 404 Not Found, status code=404
error: no match for "https://www.google.com/openshift/nodejs-e"
```

when in these cases the argument wasn't supposed to be a template file at all, it was supposed to be a valid source repo, eg:
```
$ oc new-app https://github.com/openshift/nodejs-ex --dry-run
--> Found image 1b258a4 (3 days old) in image stream "openshift/nodejs" under tag "4" for "nodejs"
```

**Comment by bparees** [comment]
On further analysis there's no obvious way to solve this.. if we got here, it means we tried to reach the url as a git repo and failed (it did not appear to be a git repo), so we fell back to trying to treat it as a template file.

since we don't know if the original failure was expected (the user gave us a legitimate url to a template) or unexpected (they tried to give us a url to a repo but we failed to reach it for some reason), it's hard to know what error message we should give them.

The right solution might be to hold the original git error and if we can't access the argument as a template file either, then report both errors ("could not reach source repo FOO.  could not load template file FOO").
